crates/valence_command/src/command.rs

```diff
@@ -0,0 +1,97 @@
+use std::any::Any;
+use std::marker::PhantomData;
+use std::ptr::NonNull;
+
+use bevy_ecs::archetype::Archetype;
+use bevy_ecs::component::Tick;
+use bevy_ecs::prelude::{Entity, EventReader};
+use bevy_ecs::system::{In, Query, System, SystemMeta, SystemParam};
+use bevy_ecs::world::unsafe_world_cell::UnsafeWorldCell;
+use bevy_ecs::world::World;
+use glam::DVec3;
+use valence_client::message::SendMessage;
+use valence_client::Client;
+use valence_core::block_pos::BlockPos;
+use valence_core::text::Text;
+
+use crate::exec::CommandExecutionEvent;
+use crate::nodes::EntityNode;
+use crate::parse::ParseResultsRead;
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum RealCommandExecutor {
+    Player(Entity),
+    Console,
```

Comment:
Why would console exist? It should be up to the user of valance to make logic to execute commands though the console. Is this used anywhere in the protocol?

Reply:
It is not covered by protocol because it has nothing to do with it, console's command executions handled by server anyway (the same as block's command executions). And I think that console can be a command executor, the same as a block.